xbmc/interfaces/json-rpc/AudioLibrary.cpp

```diff
@@ -1391,21 +1384,21 @@ JSONRPC_STATUS CAudioLibrary::RefreshAlbum(const std::string& method,
                                            const CVariant& parameterObject,
                                            CVariant& result)
 {
-  int albumID = (int)parameterObject["albumid"].asInteger();
-
   CMusicDatabase musicdatabase;
   if (!musicdatabase.Open())
     return InternalError;
 
-  //check if albumID is a valid one
+  // Check if albumID is a valid one
   CAlbum album;
+  int albumID = (int)parameterObject["albumid"].asInteger();
   if (!musicdatabase.GetAlbum(albumID, album, false))
     return InvalidParams;
 
-  std::string path = StringUtils::Format("musicdb://albums/{}/", albumID);
-
-  //execute the album refresh job
-  CMusicLibraryQueue::GetInstance().StartAlbumScan(path, true);
+  // Start rescraping additional information for the given album
+  const std::string cmd =
+      StringUtils::Format("musiclibrary.refreshalbum({})",
+                          StringUtils::Paramify(parameterObject["albumid"].asString()));
```

Comment:
For performance reasons I would introduce a variable for the albumid variant and use it as the two place where you need it in this method.

`const CVariant albumIdVariant{parameterObject["albumid"]};`